src/credentials/attestation_verifier/TestDACRevocationDelegateImpl.cpp

```diff
@@ -30,13 +31,45 @@ namespace chip {
 namespace Credentials {
 
 namespace {
+
+static constexpr uint32_t kMaxIssuerBase64Len = BASE64_ENCODED_LEN(kMaxCertificateDistinguishedNameLength);
+
 CHIP_ERROR BytesToHexStr(const ByteSpan & bytes, MutableCharSpan & outHexStr)
 {
     Encoding::HexFlags flags = Encoding::HexFlags::kUppercase;
     ReturnErrorOnFailure(BytesToHex(bytes.data(), bytes.size(), outHexStr.data(), outHexStr.size(), flags));
     outHexStr.reduce_size(2 * bytes.size());
     return CHIP_NO_ERROR;
 }
+
+CHIP_ERROR X509_PemToDer(const std::string & pemCert, MutableByteSpan & derCert)
+{
+    std::string beginMarker = "-----BEGIN CERTIFICATE-----";
+    std::string endMarker   = "-----END CERTIFICATE-----";
+
+    std::size_t beginPos = pemCert.find(beginMarker);
+    std::size_t endPos   = pemCert.find(endMarker);
+
+    // Extract content between markers
+    std::string plainB64Str = pemCert.substr(beginPos + beginMarker.length(), endPos - (beginPos + beginMarker.length()));
+
+    // Remove all newline characters '\n' and '\r'
+    plainB64Str.erase(std::remove(plainB64Str.begin(), plainB64Str.end(), '\n'), plainB64Str.end());
+    plainB64Str.erase(std::remove(plainB64Str.begin(), plainB64Str.end(), '\r'), plainB64Str.end());
```

Comment:
This is missing checks to avoid crashing if the begin/end are not found. Please make this safe and add a unit test against several invalid certs.

Reply:
Added the checks after getting marker positions, and unit tests.